forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/registrationrequest/RegistrationRequestJWTClaims.java

```diff
@@ -0,0 +1,211 @@
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.forgerock.openbanking.constants.OpenBankingConstants.SSAClaims;
+
+@Slf4j
+public class RegistrationRequestJWTClaims {
+
+    private final JWTClaimsSet claimsSet;
+    private final JWTClaimsOrigin claimsOrigin;
+    private final DynamicClientRegistrationErrorType errorType;
+
+    public RegistrationRequestJWTClaims(JWTClaimsSet claimsSet, JWTClaimsOrigin claimsOrigin){
+        this.claimsSet = claimsSet;
+        this.claimsOrigin = claimsOrigin;
+        if(claimsOrigin == JWTClaimsOrigin.REGISTRATION_REQUEST_JWT){
+            this.errorType = DynamicClientRegistrationErrorType.INVALID_CLIENT_METADATA;
+        } else {
+            this.errorType = DynamicClientRegistrationErrorType.INVALID_SOFTWARE_STATEMENT;
+        }
+    }
+
+
+    /**
+     * getRequiredStringClaim Obtains a string containing the value for a specific claim. Catches the
+     * ParseExceptions thrown by nimbusds and instead throw an DynamicClientRegistrationException.
+     *
+     * @param claimName        @String the name of the SSA claim to obtain from the ssaClaims
+     * @return A non null or empty @String containing the ssa claim value
+     * @throws DynamicClientRegistrationException when the claim specified by the claimName does not exist, or is empty.
+     */
+    public String getRequiredStringClaim(String claimName) throws DynamicClientRegistrationException {
+        Optional<String> claimValue = getOptionalStringClaim(claimName);
+        if(!stringClaimValueIsValid(claimValue)){
+            String errorString = "Claim '" + claimName + "' was not set in " + claimsOrigin + ". This claim " +
+                    "is required";
+            log.debug("getStringClaims() {}", errorString);
+            throw new DynamicClientRegistrationException(errorString, errorType);
+        }
+        return claimValue.get();
+    }
+
+    /**
+     * getRequiredStringClaim Obtains a string containing the value for a specific claim. Catches the
+     * ParseExceptions thrown by nimbusds and instead throw an DynamicClientRegistrationException.
+     *
+     * @param claimName        @String the name of the SSA claim to obtain from the ssaClaims
+     * @return An Optional<String> that will contain the value if it was present or be empty of not
+     * @throws DynamicClientRegistrationException when the claim specified by the claimName does not exist, or is empty.
+     */
+    public Optional<String> getOptionalStringClaim(String claimName) throws DynamicClientRegistrationException {
+        try {
+            return Optional.ofNullable(claimsSet.getStringClaim(claimName));
+        } catch (ParseException e) {
+            String errorString =
+                    "Failed to parse claim '" + claimName + "' from claim set in " + claimsOrigin.toString();
+            log.debug("getStringClaims() {}", errorString, e);
```

Comment:
Could be better log.error or log.warn instead of log.debug? because contains the error message.